### DISCUSSION TITLE
[Resolved] Changes in SIGMET in Amendment 79

### DISCUSSION THREAD

**Discussion by blchoy** [wmo-im/iwxxm#267] [Resolution]
I would like to continue the discussion on [validation of changes for SIGMET](https://github.com/wmo-im/iwxxm/wiki/Validation:-IWXXM-2021_2RC1#sigmet-follow-up-in250) here.

---

Looking at the examples in the latest [APAC Regional SIGMET Guide](https://www.icao.int/APAC/Documents/edocs/APAC%20Regional%20SIGMET%20Guide%208th%20edition%20Nov%202020%20-%20final.pdf), I would like to propose the following changes to the SIGMET package:

### 1. It only needs to handle only one tropical cyclone and one volcano
This is in accordance to the revised Note 28 (renamed to Note 29 in Amendment 79) of the SIGMET template.

### 2. Multiple VA or CB clouds can be described through the use of AND.  Each of them is to be preceded by an observed or forecast time
This is in accordance to Notes 20 and 21 in Amendment 79.

### 3. TAC-to-XML Guidance may need to be updated on possible nil OBS/FCST time
Even though iwxxm:analysis//iwxxm:phenomenonTime can be empty, we may want to include a nilReason to indicate it is intentionally left blank.

---

The information model for TC SIGMET in IWXXM 3.0 which was based on the requirements in Am78 looks something like (similar for VA SIGMET):

```
<iwxxm:TropicalCycloneSIGMET>
    <iwxxm:issueTime/>
    <iwxxm:issuingAirTrafficServicesUnit/>
    <iwxxm:originatingMeteorologicalWatchOffice/>
    <iwxxm:issuingAirTrafficServicesRegion/>
    <iwxxm:sequenceNumber/>
    <iwxxm:validPeriod/>
    <iwxxm:phenomenon/>

    <iwxxm:analysis>
        <iwxxm:TropicalCycloneSIGMETEvolvingConditionCollection>
            <iwxxm:phenomenonTime/>
            <iwxxm:member>
                <iwxxm:SIGMETEvolvingCondition/>
            </iwxxm:member>
            <iwxxm:member/>    <-- More members as required
            <iwxxm:tropicalCyclonePosition/>
        </iwxxm:TropicalCycloneSIGMETEvolvingConditionCollection>
    </iwxxm:analysis>
    <iwxxm:analysis/>   <-- for another TC

    <iwxxm:forecastPositionAnalysis>
        <iwxxm:TropicalCycloneSIGMETPositionCollection>
            <iwxxm:phenomenonTime/>
            <iwxxm:member>
                <iwxxm:SIGMETPosition/>
            </iwxxm:member>
            <iwxxm:member/>    <-- More members as required
            <iwxxm:tropicalCyclonePosition/>
        </iwxxm:TropicalCycloneSIGMETPositionCollection>
    </iwxxm:forecastPositionAnalysis>
    <iwxxm:forecastPositionAnalysis/>    <-- for another TC
 
    <iwxxm:tropicalCyclone>
        <metce:TropicalCyclone/>
    </iwxxm:tropicalCyclone>
    <iwxxm:tropicalCyclone/>    <-- for another TC

</iwxxm:TropicalCycloneSIGMET>
```

---

Am79 to Annex 3 did not give any example on a TC with multiple CBs.

The [APAC Regioinal SIGMET Guide](https://www.icao.int/APAC/Documents/edocs/APAC%20Regional%20SIGMET%20Guide%208th%20edition%20Nov%202020%20-%20final.pdf) did give one on VA and another one on TC SIGMET:

Example 8 (VA SIGMET)
```
YUDD SIGMET 2 VALID 101200/101800 YUSO–
YUDD SHANLON FIR VA ERUPTION MT ASHVAL PSN N4315 E02115 VA CLD OBS AT 
1200Z WI N4315 E02115 – N4345 E02145 - N4330 E02215 – N4245 E02230 –
N4230 E02145 - N4315 E02115 FL250/370 NC FCST AT 1800Z WI N4330 E02215 
– N4315 E02345 – N4145 E02315 – N4230 E02200 - N4330 E02215 AND OBS AT 
1200Z WIN4200 E02115 – N4217 E02130 – N4145 E02200 – N4130 E02130 –
N4200 E02115 FL150/300 NC FCST AT 1800Z WI N4200 E02145 – N4145 E02215 
– N4100 E02215 - N4130 E02130 - N4200 E02145=
```

Example 10 (TC SIGMET):
```
YUDD SIGMET 3 VALID 251230/251830 YUSO-
YUDD SHANLON FIR TC GLORIA PSN N22 W061 CB OBS AT 1200Z WI 050KM OF TC 
CENTRE TOP FL500 INTSF FCST AT 1800Z TC CENTRE PSN N2230 W06230 CB WI 
060KM OF TC CENTRE AND OBS AT 1220Z WI N2230 W06045 - N2230 W06015 - 
N2200 W06030 - N2230 W06045 TOP ABV FL500 FCST AT 1800Z WI N2300 W06200 
- N2300 W06130 -N2230 W06145 -N2300 W06200=
```

---

### Do nothing approach

If we do nothing, it is still possible to describe the VA SIGMET example.  The only drawback is that all VA clouds in OBS/FCST will have to squeeze into `<iwxxm:analysis>` and those in FCST into `<iwxxm:forecastPositionAnalysis>` and the connection of OBS/FCST and FCST of individual VA cloud will be lost.

For the TC SIGMET example as the OBS/FCST time of the second CB was different from the main one there is no way of representing this situation.

### Active approach: Change in accordance to the examples in the APAC SIGMET Guide

This will require a major change in information model from describing

OBS/FCST DTG1 CLD_A CLD_B CLD_C FCST DTG2 CLD_A CLD_B CLD_C

to

OBS/FCST DTG1A CLD_A FCST DTG2A **AND** OBS/FCST DTG1B CLD_B FCST DTG2B CLDB **AND** OBS/FCST DTG1C CLD_C FCST DTC2C CLD_C

where DTG1A, DTG1B and DTG1C can be different, as well as DTG2A, DTG2B and DTG2C.

---

E-mail exchange on views in supporting multiple OBS/FCST time can be found [here](https://github.com/wmo-im/iwxxm-translation/files/6581970/email_exchange.pdf).  To continue with this discussion, I will separately post a new proposed change to IWXXM in accordance to the active approach.

**Comment by blchoy** [comment]
The following UML diagrams indicate proposed changes of the active approach:

![Context Diagram SIGMET](https://user-images.githubusercontent.com/30064374/120434953-acc1b080-c3af-11eb-9bfc-0873b34bc50a.png)
![Context Diagram SIGMET Analysis](https://user-images.githubusercontent.com/30064374/120434990-ba773600-c3af-11eb-9585-bb69071e2e86.png)
![Context Diagram SIGMET Forecast Position Analysis](https://user-images.githubusercontent.com/30064374/120436253-49388280-c3b1-11eb-9226-cd57abef04b7.png)

A TC SIGMET instance will become:

```
<iwxxm:TropicalCycloneSIGMET>
    <iwxxm:issueTime/>
    <iwxxm:issuingAirTrafficServicesUnit/>
    <iwxxm:originatingMeteorologicalWatchOffice/>
    <iwxxm:issuingAirTrafficServicesRegion/>
    <iwxxm:sequenceNumber/>
    <iwxxm:validPeriod/>
    <iwxxm:phenomenon/>

    <iwxxm:analysis>
        <iwxxm:TropicalCycloneSIGMETEvolvingConditionCollection>
            <iwxxm:phenomenonTime/>
            <iwxxm:member>
                <iwxxm:SIGMETEvolvingCondition/>
            </iwxxm:member>
            <iwxxm:tropicalCyclonePosition/>
        </iwxxm:TropicalCycloneSIGMETEvolvingConditionCollection>
    </iwxxm:analysis>

    <iwxxm:forecastPositionAnalysis>
        <iwxxm:TropicalCycloneSIGMETPositionCollection>
            <iwxxm:phenomenonTime/>
            <iwxxm:member>
                <iwxxm:SIGMETPosition/>
            </iwxxm:member>
            <iwxxm:tropicalCyclonePosition/>
        </iwxxm:TropicalCycloneSIGMETPositionCollection>
    </iwxxm:forecastPositionAnalysis>
 
    <iwxxm:supplementaryAnalysisCollection>
        <iwxxm:analysisAndForecastPositionAnalysis>
            <iwxxm:analysis>
                <iwxxm:SIGMETEvolvingConditionCollection>
                    <iwxxm:phenomenonTime/>
                    <iwxxm:member>
                        <iwxxm:SIGMETEvolvingCondition/>
                    </iwxxm:member>
                </iwxxm:SIGMETEvolvingConditionCollection>
            </iwxxm:analysis>
            <iwxxm:forecastPositionAnalysis>
                <iwxxm:SIGMETPositionCollection>
                    <iwxxm:phenomenonTime/>
                    <iwxxm:member>
                        <iwxxm:SIGMETPosition/>
                    </iwxxm:member>
                </iwxxm:SIGMETPositionCollection>
            </iwxxm:forecastPositionCollection>
        </iwxxm:analysisAndForecastPositionAnalysis>
    </iwxxm:supplementaryAnalysisCollection>
    <iwxxm:supplementaryAnalysisCollection/>    <-- If more CBs

    <iwxxm:tropicalCyclone>
        <metce:TropicalCyclone/>
    </iwxxm:tropicalCyclone>

</iwxxm:TropicalCycloneSIGMET>
```

And a VA SIGMET will become:

```
<iwxxm:VolcanicAshSIGMET>
    <iwxxm:issueTime/>
    <iwxxm:issuingAirTrafficServicesUnit/>
    <iwxxm:originatingMeteorologicalWatchOffice/>
    <iwxxm:issuingAirTrafficServicesRegion/>
    <iwxxm:sequenceNumber/>
    <iwxxm:validPeriod/>
    <iwxxm:phenomenon/>

    <iwxxm:analysis>
        <iwxxm:SIGMETEvolvingConditionCollection>
            <iwxxm:phenomenonTime/>
            <iwxxm:member>
                <iwxxm:SIGMETEvolvingCondition/>
            </iwxxm:member>
        </iwxxm:SIGMETEvolvingConditionCollection>
    </iwxxm:analysis>

    <iwxxm:forecastPositionAnalysis>
        <iwxxm:SIGMETPositionCollection>
            <iwxxm:phenomenonTime/>
            <iwxxm:member>
                <iwxxm:SIGMETPosition/>
            </iwxxm:member>
        </iwxxm:SIGMETPositionCollection>
    </iwxxm:forecastPositionAnalysis>
 
    <iwxxm:supplementaryAnalysisCollection>
        <iwxxm:analysisAndForecastPositionAnalysis>
            <iwxxm:analysis>
                <iwxxm:SIGMETEvolvingConditionCollection>
                    <iwxxm:phenomenonTime/>
                    <iwxxm:member>
                        <iwxxm:SIGMETEvolvingCondition/>
                    </iwxxm:member>
                </iwxxm:SIGMETEvolvingConditionCollection>
            </iwxxm:analysis>
            <iwxxm:forecastPositionAnalysis>
                <iwxxm:SIGMETPositionCollection>
                    <iwxxm:phenomenonTime/>
                    <iwxxm:member>
                        <iwxxm:SIGMETPosition/>
                    </iwxxm:member>
                </iwxxm:SIGMETPositionCollection>
            </iwxxm:forecastPositionCollection>
        </iwxxm:analysisAndForecastPositionAnalysis>
    </iwxxm:supplementaryAnalysisCollection>
    <iwxxm:supplementaryAnalysisCollection/>    <-- If more VA clouds

    <iwxxm:eruptingVolcano>
        <metce:Volcano/>
    </iwxxm:eruptingVolcano>

</iwxxm:VolcanicAshSIGMET>
```

**Comment by blchoy** [comment]
I am now opening up the floor for discussion.

**Comment by jkorosi** [comment]
After several hours of analysis, I realized that observation time/movement/intensity/timeIndicator issues can be solved by wrapping **<iwxxm:analysis>** and **<iwxxm:forecastPositionAnalysis>** to **<iwxxm:analysisAndForecastPositionAnalysis>** in the same way how Choy defines **<iwxxm:supplementaryAnalysisCollection>**.
Each OBS+FCST pair has to be define inside **<iwxxm:analysisAndForecastPositionAnalysis>**.

We "just" need to clarify that:
1. Only the TC CENTRE shall be defined in **<iwxxm:analysisAndForecastPositionAnalysis>**.
2. Therefore **<iwxxm:tropicalCyclonePosition/>** shall be removed from the model.
3. All CB clouds have to be defined in **<iwxxm:supplementaryAnalysisCollection>**.
4. Each OBS+FCST pair shall be wrapped in **<iwxxm:analysisAndForecastPositionAnalysis>**. In other words, the VA SIGMET will be the only one that will repeat this element when it is necessary.
5. At the end of the day, we can remove also:
**<iwxxm:VolcanicAshSIGMETEvolvingConditionCollection/>**
**<iwxxm:VolcanicAshSIGMETPositionCollection>**
**<iwxxm:TropicalCycloneSIGMETPositionCollection>**
**<iwxxm:TropicalCycloneSIGMETPositionCollection>**
because they will be equal to **<iwxxm:SIGMETEvolvingConditionCollection/>** and **<iwxxm:SIGMETPositionCollection>**. We will unify all SIGMET types!

In addition to TAC, we will be able to report that one cloud will break up into several parts by repetition of **<iwxxm:member>**.

The VA SIGMET should look like
```xml
<iwxxm:VolcanicAshSIGMET>
  <iwxxm:issueTime/>
  <iwxxm:issuingAirTrafficServicesUnit/>
  <iwxxm:originatingMeteorologicalWatchOffice/>
  <iwxxm:issuingAirTrafficServicesRegion/>
  <iwxxm:sequenceNumber/>
  <iwxxm:validPeriod/>
  <iwxxm:phenomenon/>
  <!-- first analysis and forecast pair (with forecast)-->
  <iwxxm:analysisAndForecastPositionAnalysis>
    <iwxxm:analysis>
      <iwxxm:SIGMETEvolvingConditionCollection timeIndicator="OBSERVATION">
        <iwxxm:phenomenonTime/>
        <iwxxm:member>
          <iwxxm:SIGMETEvolvingCondition intensityChange="NO_CHANGE">
            <iwxxm:geometry/>
          </iwxxm:SIGMETEvolvingCondition>
        </iwxxm:member>
      </iwxxm:SIGMETEvolvingConditionCollection>
    </iwxxm:analysis>
    <iwxxm:forecastPositionAnalysis>
      <iwxxm:SIGMETPositionCollection>
        <iwxxm:phenomenonTime/>
        <iwxxm:member>
          <iwxxm:SIGMETPosition approximateLocation="true">
            <iwxxm:geometry/>
          </iwxxm:SIGMETPosition>
        </iwxxm:member>
      </iwxxm:SIGMETPositionCollection>
    </iwxxm:forecastPositionAnalysis>
  <iwxxm:analysisAndForecastPositionAnalysis>
  <!-- next analysis and forecast pair (with movement) -->
  <iwxxm:analysisAndForecastPositionAnalysis>
    <iwxxm:analysis>
      <iwxxm:SIGMETEvolvingConditionCollection timeIndicator="OBSERVATION">
        <iwxxm:phenomenonTime/>
        <iwxxm:member>
          <iwxxm:SIGMETEvolvingCondition intensityChange="NO_CHANGE">
            <iwxxm:geometry/>
            <iwxxm:directionOfMotion/>
            <iwxxm:speedOfMotion/>
          </iwxxm:SIGMETEvolvingCondition>
        </iwxxm:member>
      </iwxxm:SIGMETEvolvingConditionCollection>
    </iwxxm:analysis>
  <iwxxm:analysisAndForecastPositionAnalysis>
  <!-- volcano definition -->
  <iwxxm:eruptingVolcano/>
</iwxxm:VolcanicAshSIGMET>
```

The TC SIGMET should look like

```xml
<iwxxm:TropicalCycloneSIGMET>
  <iwxxm:issueTime/>
  <iwxxm:issuingAirTrafficServicesUnit/>
  <iwxxm:originatingMeteorologicalWatchOffice/>
  <iwxxm:issuingAirTrafficServicesRegion/>
  <iwxxm:sequenceNumber/>
  <iwxxm:validPeriod/>
  <iwxxm:phenomenon/>
  <!-- report TC CENTRE analysis and forecasts with movement and intensity -->
  <iwxxm:analysisAndForecastPositionAnalysis>
    <iwxxm:analysis>
      <iwxxm:SIGMETEvolvingConditionCollection>
        <iwxxm:phenomenonTime/>
        <iwxxm:member>
          <iwxxm:SIGMETEvolvingCondition/> <!-- contains only one point where is the TC CENTRE -->
        </iwxxm:member>
      </iwxxm:SIGMETEvolvingConditionCollection>
    </iwxxm:analysis>
    <iwxxm:forecastPositionAnalysis>
      <iwxxm:SIGMETPositionCollection>
        <iwxxm:phenomenonTime/>
        <iwxxm:member>
          <iwxxm:SIGMETPosition/>
        </iwxxm:member>
      </iwxxm:SIGMETPositionCollection>
    </iwxxm:forecastPositionAnalysis>
  </iwxxm:analysisAndForecastPositionAnalysis>
  <!-- report related CB clouds analysis and forecasts without movement and intensity -->
  <iwxxm:supplementaryAnalysisCollection>
    <iwxxm:analysisAndForecastPositionAnalysis>
      <iwxxm:analysis>
        <iwxxm:SIGMETEvolvingConditionCollection>
          <iwxxm:phenomenonTime/>
          <iwxxm:member>
            <iwxxm:SIGMETEvolvingCondition/>
          </iwxxm:member>
        </iwxxm:SIGMETEvolvingConditionCollection>
      </iwxxm:analysis>
      <iwxxm:forecastPositionAnalysis>
        <iwxxm:SIGMETPositionCollection>
          <iwxxm:phenomenonTime/>
          <iwxxm:member>
            <iwxxm:SIGMETPosition/>
          </iwxxm:member>
        </iwxxm:SIGMETPositionCollection>
      </iwxxm:forecastPositionCollection>
    </iwxxm:analysisAndForecastPositionAnalysis>
  </iwxxm:supplementaryAnalysisCollection>
  <!-- report next related CB clouds analysis and forecasts without movement and intensity -->
  <iwxxm:supplementaryAnalysisCollection/>
  <!-- report tropical cyclone -->
  <iwxxm:tropicalCyclone>
    <metce:TropicalCyclone/>
  </iwxxm:tropicalCyclone>
</iwxxm:TropicalCycloneSIGMET>
```

**Reply by blchoy** [reply]
Sorry for not following up earlier.  I was monitoring the discussions at METP/5 and separate discussions among members of WG-MIE, WG-MOG and APAC teams on the TC example with multiple OBS time before and after AND.  With regard to your proposals I have a couple of comments and questions for discussion at the upcoming meeting on 24 Jun:

1. Your suggestion to also use `<iwxxm:analysisAndForecastPositionAnalysis>` in the main part of the SIGMET message will also require the IWXXM version of WS SIGMET to carry this block.  This is good from a consistency (WS, WV and WC SIGMETs) view point but is less desirable from a minimum change view point.  I have no strong views on this and I am looking forward to a consensus within the team.
2. With regard to your VA proposal it is interesting to see you indicating the possibility of having different representations within `<iwxxm:analysisAndForecastPositionAnalysis>` before and after AND.  Example 8 of the APAC SIGMET guide hinted that they will be the same but that was just an example.  Have you come across real life examples?  I think it is sensible to keep this flexibility but should not publicize this kind of usage as the reporting practices shall be determined by ICAO teams, not us.
3. With regard to TC SIGMET I still prefer my previous proposal as the TC centre and the first CB should be considered as a single object (or speaking the other way round, an object with two properties, viz centre position and CB cloud coverage).  Putting TC centre and the first CB cloud into two separate `<iwxxm:analysisAndForecastPositionAnalysis>` may introduce unnecessary de-linkage of the centre and most significant CB.
4. I agree to keep the possibility of using `<iwxxm:member>` to cater for possible further segregation of clouds, but again let's not advertise its possible use as this may not be the intended use of the ICAO teams.

**Reply by amilan17** [reply]
all agreed to these decisions.

**Comment by blchoy** [comment]
I would like to summarize the discussion of this topic in METP/5 below:

1. There was no time to bring up this topic in METP/5, which covered whether we should use multiple OBS/FCST times in a TC/VA SIGMET with AND, and how the representation should be with AND as described in ICAO Doc 8896.
2. A separate discussion among members of the WG-MIE, WG-MOG and APAC teams reached the consensus including:
   * They were in favour of the representation in the APAC SIGMET guide:
     ```
     OBS CB cloud1        Intensity         FCST CB cloud1
     AND
     OBS CB cloud2        FCST CB cloud2
     ```
   * The representation in ICAO Doc 8896 as shown below does not comply with Annex 3 and should be corrected:
      ```
      OBS CB cloud1         AND         OBS CB cloud2         Movement         Intensity
      FCST CB cloud1        AND         FCST CB2 cloud
      ```
   * Setting the same OBS/FCST times for multiple CB areas would be the most user-friendly.  Thus the example in the APAC SIGMET guide should be updated.  However, as Annex 3 does allow the flexibility of different OBS/FCST time, so the follow up action could either be (i) add this capability in IWXXM SIGMET, or (ii) wait for the next Annex 3 amendment to revert back to the same OBS/FCST time.

Views are most welcomed.

**Comment by blchoy** [comment]
Thank you all for your contribution and I am glad to close this discussion at this point.
